pkg/lso/localvolumediscovery.go

```diff
@@ -0,0 +1,168 @@
+// and a localVolumeDiscovery definition.
+type LocalVolumeDiscoveryBuilder struct {
+	// localVolumeDiscovery definition, used to create the localVolumeDiscovery object.
+	Definition *lsoV1alpha1.LocalVolumeDiscovery
+	// Created localVolumeDiscovery object.
+	Object *lsoV1alpha1.LocalVolumeDiscovery
+	// Used in functions that define or mutate localVolumeDiscovery definition. errorMsg is processed
+	// before the localVolumeDiscovery object is created
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewLocalVolumeDiscoveryBuilder creates new instance of LocalVolumeDiscoveryBuilder.
+func NewLocalVolumeDiscoveryBuilder(apiClient *clients.Settings, name, nsname string) *LocalVolumeDiscoveryBuilder {
+	glog.V(100).Infof("Initializing new localVolumeDiscovery structure with the following params: name: "+
+		"%s, namespace: %s", name, nsname)
+
+	builder := LocalVolumeDiscoveryBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeDiscovery{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Discover fetches existing localVolumeDiscovery from cluster.
+func (builder *LocalVolumeDiscoveryBuilder) Discover() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Pulling existing localVolumeDiscovery with name %s under namespace %s from cluster",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	lvd := &lsoV1alpha1.LocalVolumeDiscovery{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, lvd)
+
+	builder.Object = lvd
+
+	return err
+}
+
+// Exists checks whether the given localVolumeDiscovery exists.
+func (builder *LocalVolumeDiscoveryBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if localVolumeDiscovery %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	err := builder.Discover()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// IsDiscovering check if the localVolumeDiscovery is Discovering.
+func (builder *LocalVolumeDiscoveryBuilder) IsDiscovering() (bool, error) {
+	if valid, err := builder.validate(); !valid {
+		return false, err
+	}
+
+	glog.V(100).Infof("Verify localVolumeDiscovery %s in namespace %s is successful",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	csvFound := builder.Exists()
```

Comment:
I'm not familuar with lso operator. Can you share what is  csvFound ?

Reply:
I remove this name to avoid confusion